storages/indexeddb-storage/src/lib.rs

```diff
@@ -0,0 +1,75 @@
+#![cfg(target_arch = "wasm32")]
+#![allow(clippy::future_not_send)]
+
+mod alter_table;
+mod index;
+mod key;
+mod query;
+mod storage_error;
+mod store;
+mod store_mut;
+mod transaction;
+
+use {
+    gluesql_core::result::Result,
+    idb::{Database, Factory, ObjectStoreParams},
+};
+
+pub struct IndexeddbStorage {
+    database: Database,
+    id_ctr: u32,
+}
+
+const SCHEMA_STORE: &str = "schemas";
+const DATA_STORE: &str = "data";
```

Comment:
I thought about that as well, but unfortunately, stores can only be modified during database creation in the onupgradeneeded callback ([source](https://stackoverflow.com/a/58851804)). Unless the whole database should be recreated for each table creation or deletion, we cannot map schemas to stores.

Reply:
Ah, ok transaction issue is not the only one that can make this issue tough.
I hope that we can find the solution for this because not only the ones that I mentioned, but also we can have a powerful new feature if we succeed to make each table data to have its own object store.
I'm currently working on providing schemaless data support which allows us to query without providing schema infomation using CREATE TABLE.
This means a lot in indexeddb storage.
With schemaless support, users can use GlueSQL for data stored in indexeddb that no matter the data is created by GlueSQL or not.

I'll also think about how to deal this issue.

Reply:
Should we shelve IndexedDB and this PR until that's implemented then? Do you have an ETA? I or anybody who needs it in the meantime can just use what's in this PR as a custom storage in their own projects, so as long as it exists, it doesn't need to officially become part of the crate.

Reply:
> Should we shelve IndexedDB and this PR until that's implemented then? Do you have an ETA?

Nope, that is just for your information. you can think the schemaless related ones independent with any other ones.
I'll also check that how we can handle `onupgradeneeded` event properly.

Reply:
Alright, please let me know once you've got a better idea then, and we can apply necessary changes. :)